api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -417,7 +417,7 @@ private void setObsReferenceRange(Obs obs) {
 		
 		ConceptNumeric conceptNumeric = Context.getConceptService().getConceptNumeric(obs.getConcept().getId());
 
-		if (conceptNumeric != null) {
+		if (conceptNumeric != null && obs.getId() == null) {
```

Comment:
Wouldn't it be better to instead do the check whenever you are about to call setObsReferenceRange()?